linkerd/tracing/src/access_log.rs

```diff
@@ -0,0 +1,166 @@
+use std::{fmt, sync::Arc};
+use tracing::{field, span, Id, Level, Metadata, Subscriber};
+use tracing_appender::non_blocking::WorkerGuard;
+use tracing_subscriber::{
+    field::RecordFields,
+    filter::{Directive, FilterFn, Filtered},
+    fmt::{format, FormatFields, FormattedFields},
+    layer::{Context, Layer},
+    registry::LookupSpan,
+};
+
+const ENV_ACCESS_LOG: &str = "LINKERD2_PROXY_ACCESS_LOG";
+
+pub const TRACE_TARGET: &str = "_access_log";
+
+#[derive(Clone, Debug)]
+pub struct Guard(Arc<WorkerGuard>);
+
+pub(super) type AccessLogLayer<S> = Filtered<Writer, FilterFn, S>;
+
+pub(super) struct Writer<F = ApacheCommon> {
+    formatter: F,
+}
+
+#[derive(Default)]
+pub(super) struct ApacheCommon {
+    _p: (),
+}
+
+struct ApacheCommonVisitor<'writer> {
+    res: fmt::Result,
+    writer: format::Writer<'writer>,
+}
+
+pub(super) fn build<S>() -> Option<(AccessLogLayer<S>, Directive)>
+where
+    S: Subscriber + for<'span> LookupSpan<'span>,
+{
+    // Is access logging enabled?
+    let _ = std::env::var(ENV_ACCESS_LOG).ok()?;
```

Comment:
Or if we can't move this into `linkerd_app`, could we at least make this part of `linkerd_tracing::Settings::from_env`?

Reply:
i don't remember off the top of my head, but i think there might have been a reason we defined the logging env vars here instead, maybe to avoid a cyclic dependency? but i'll take a look and see if that's possible!

Reply:
can definitely at least move it into `from_env` though.

Reply:
forgot to reply to this earlier, but, yeah, the reason the logging env vars aren't in `linkerd-app` is because `linkerd-app` depends on `linkerd-tracing`, so it would create a cyclical dependency (i double checked). i did move it into `Settings::from_env` though.